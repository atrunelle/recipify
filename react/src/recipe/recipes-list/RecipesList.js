import React, { Component } from 'react';
import { connect } from 'react-redux';

class RecipesList extends Component {
  render() {
    if (!this.props.recipes.length) return (<div>
      <p>You don't have any recipes yet</p>
      <a>Create recipes</a>
    </div>);
    return (
      <div></div>
    );
  }
}

const mapStateToProps = (state) => ({
	recipes: state.recipes.recipes,
});

export default connect(mapStateToProps)(RecipesList);