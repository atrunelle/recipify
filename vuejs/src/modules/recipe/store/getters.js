import * as types from './types';

export default {
  [types.GET_RECIPE]: (state) => state.recipeIngredients,
  [types.GET_RECIPES]: (state) => state.recipes,
};
