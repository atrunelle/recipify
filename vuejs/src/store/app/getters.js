import * as types from './types';

export default {
  [types.GET_ALERT]: (state) => state.alert,
};
