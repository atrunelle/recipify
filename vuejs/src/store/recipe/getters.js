import nutritionService from '@/core/nutrition.service';

export default {
  totalWeight: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.totalWeight + sum;
    }, 0);
  },

  totalCalories: (state) => {
    return state.ingredients.reduce((sum, ingredient) => {
      return ingredient.nutrients.calories + sum;
    }, 0);
  },

  totalNutrients: (state, getters) => {
    return nutritionService
      .getTotalNutrients(state.ingredients, getters.totalCalories);
  },
};
