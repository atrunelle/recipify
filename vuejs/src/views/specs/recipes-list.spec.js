import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import RecipesList from '../recipes-list';
import getters from '@/modules/recipe/store/getters';

describe('Component: recipes list', () => {
  let component;
  let localVue = createLocalVue();

  localVue.use(Vuex);

  beforeEach(() => {
    const recipeStore = {
      namespaced: true,
      state: {
        recipes: [{
          name: 'recipe 1',
          ingredients: [{
            name: 'apple',
            nutrients: {
              calories: 100,
              totalWeight: 200,
            },
          }],
        },
        ],
      },
      getters,
    };

    const store = new Vuex.Store({
      modules: {
        recipe: recipeStore,
      },
    });

    component = mount(RecipesList, {
      localVue,
      store,
    });
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
