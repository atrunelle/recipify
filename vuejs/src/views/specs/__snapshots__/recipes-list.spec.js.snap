// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: recipes list should match snapshot 1`] = `
<div class="container grid-list-md">
  <div class="layout row wrap">
    <div class="flex md4">
      <div class="v-card v-sheet theme--light" style="height: 100%;">
        <div class="v-card__title v-card__title--primary">
          <h2 class="headline">recipe 1</h2>
        </div>
        <div role="list" class="v-list theme--light">
          <div role="listitem">
            <div class="v-list__tile theme--light">
              <div class="v-list__tile__content">
                <div class="v-list__tile__title">apple 100cal for 200gr </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>
`;
